ansible/tasks/create_aro_cluster.yaml

```diff
@@ -0,0 +1,134 @@
+---
+- name: create_aro_cluster | Set up a jumphost for private clusters
+  when: apiserver_visibility is defined and apiserver_visibility == "Private"
+  ansible.builtin.include_tasks:
+    file: ../tasks/create_jumphost.yaml
+
+# TODO: Create clusters with azure_rm_openshiftmanagedcluster if possible
+# - name: Create cluster service principal
+#   azure.azcollection.azure_rm_adserviceprincipal:
+#     delegate_to: localhost
+#     app_id: "{{ name | to_uuid }}"
+#     tenant: "{{ sub_info.tenant_id }}"
+#   register: csp_info
+# - ansible.builtin.debug: var=csp_info
+# - name: Create cluster service principal
+#   ansible.builtin.command:
+#     delegate_to: localhost
+#     argv: [
+#       "az", "ad", "sp", "create-for-rbac",
+#       "-n", "{{ name }}-sp",
+#       "--role", "contributor",
+#       "--scopes", "{{ rg_info.state.id }}"
+#     ]
+#   register: sp_info
+# - ansible.builtin.debug: var=sp_info
+# - name: Create aro cluster
+#   azure.azcollection.azure_rm_openshiftmanagedcluster:
+#     delegate_to: localhost
+#     name: "{{ name }}"
+#     resource_group: "{{ resource_group }}"
+#     location: "{{ location }}"
+#     service_principal_profile:
+#       client_id: "{{ csp_info.state.client_id }}"
+#       client_secret: "{{ csp_info.state.client_secret }}"
+#     worker_profiles:
+#       - name: "worker"
+#         vm_size: "Standard_D4s_v3"
+#         subnet_id: "{{ worker_subnet_state.state.id }}"
+#     master_profile:
+#       vm_size: "Standard_D4s_v3"
+#       subnet_id: "{{ master_subnet_state.state.id }}"
+- name: create_aro_cluster | Check if cluster already exists
+  ansible.builtin.command:
+    argv: ["az", "aro", "show", "--name={{ name }}", "--resource-group={{ resource_group }}", "-o=yaml"]
+  failed_when: "'provisioningState: Succeeded' not in aro_existing_cluster.stdout"
+  delegate_to: localhost
+  register: aro_existing_cluster
+  ignore_errors: true
+- name: create_aro_cluster | Create aro cluster
+  when: aro_existing_cluster is not success
```

Comment:
This condition wouldn't be enough to verify if the cluster already exists. According to this, if we have a cluster  already existing with `provisioningState: Failed`, it will still initiate the cluster creation which will eventually fail. 

I think if we get any result to the `az aro show` command for the cluster, we could consider the cluster to be present ?

Reply:
This is somewhat intentional, since I haven't decided what to do if we encounter a non-successful cluster here. Delete it and recreate it? Fail out? What if it's not in a terminal state?